aggregator/src/aggregator.rs

```diff
@@ -1942,99 +1942,104 @@ impl VdafOps {
-                                task.id(),
-                                report_share_data.report_share.metadata().id(),
-                                aggregation_job.aggregation_parameter(),
-                                aggregation_job.id(),
-                            ),
-                            Q::get_conflicting_aggregate_share_jobs::<SEED_SIZE, C, A>(
-                                tx,
-                                &vdaf,
-                                task.id(),
-                                req.batch_selector().batch_identifier(),
-                                report_share_data.report_share.metadata()
-                            ),
-                        )?;
-
-                        if report_aggregation_exists {
-                            report_share_data.report_aggregation =
-                                report_share_data.report_aggregation
+        Ok(
+            datastore
+                .run_tx("aggregate_init", |tx| {
+                    let vdaf = vdaf.clone();
+                    let task = Arc::clone(&task);
+                    let req = Arc::clone(&req);
+                    let aggregation_job = Arc::clone(&aggregation_job);
+                    let mut report_share_data = report_share_data.clone();
+                    let interval_per_batch_identifier = Arc::clone(&interval_per_batch_identifier);
+                    let accumulator = Arc::clone(&accumulator);
+
+                    Box::pin(async move {
+                        for report_share_data in &mut report_share_data {
+                            // Verify that we haven't seen this report ID and aggregation parameter
+                            // before in another aggregation job, and that the report isn't for a batch
+                            // interval that has already started collection.
+                            let check_fut = tx
+                                .check_other_report_aggregation_exists::<SEED_SIZE, A>(
+                                    task.id(),
+                                    report_share_data.report_share.metadata().id(),
+                                    aggregation_job.aggregation_parameter(),
+                                    aggregation_job.id(),
+                                );
+                            let conflict_fut =
+                                Q::get_conflicting_aggregate_share_jobs::<SEED_SIZE, C, A>(
+                                    tx,
+                                    &vdaf,
+                                    task.id(),
+                                    req.batch_selector().batch_identifier(),
+                                    report_share_data.report_share.metadata(),
+                                );
+                            // These futures intentionally have short names and are defined separately
+                            // from the `try_join!` macro due to a `rustfmt` bug.
```

Comment:
FWIW, this isn't so much a bug in rustfmt as a limitation inherent to the language. Tokens inside a macro follow a much looser syntax, and in this context, many of the formatting rules are not safe to apply mechanically. A common example is `assert_eq!()` and `trailing_comma = "Vertical"`. If one were to break up an `assert_eq!()` into multiple lines, the usual rule would have you add a trailing comma after the second argument, but none of `assert_eq!()`'s rules matches that. If there's a comma after the second argument, it expects there to be at least a third message argument.

Reply:
I was thinking it a bug since the one `try_join!` breaks the ability to format the entire long closure. Point still valid though, I'll `s/bug/limitation/`'.

Reply:
Another aspect of rustfmt is that it will do nothing to an entire "chain" if there's no way to format it that satisfies all rules. We've run into this before with long string literals or a combination of indentation, long method names, and a big turbofish.